src/Codeception/Module/Db.php

```diff
@@ -801,8 +801,13 @@ private function addInsertedRow(string $table, array $row, $id): void
         $primaryKey = $this->_getDriver()->getPrimaryKey($table);
         $primary = [];
         if ($primaryKey !== []) {
-            if ($id && count($primaryKey) === 1) {
-                $primary [$primaryKey[0]] = $id;
+            $filledKeys = array_intersect($primaryKey, array_keys($row));
+            $primaryKeyIsFilled = count($filledKeys) === count($primaryKey);
+
+            if ($primaryKeyIsFilled) {
+                $primary = array_intersect_key($row, array_flip($primaryKey));
+            } elseif ($id && count($primaryKey) === 1) {
+                $primary[$primaryKey[0]] = $id;
```

Comment:
myisam table with auto_increment on second field of primary key would still fail:
https://dev.mysql.com/doc/mysql-tutorial-excerpt/5.7/en/example-auto-increment.html#example-auto-increment-myisam-notes

Reply:
Yes. Also the purpose of this PR is not to address this issue.

Reply:
I felt bad and added support for the case you brought up 🙃